zerver/views/streams.py

```diff
@@ -416,6 +420,13 @@ def update_stream_backend(
             check_stream_name_available(user_profile.realm, new_name)
         do_rename_stream(stream, new_name, user_profile)
 
+    if folder_id is not Missing:
+        assert not isinstance(folder_id, MissingType)
```

Comment:
```diff
-    if folder_id is not Missing:
-        assert not isinstance(folder_id, MissingType)
+    if not isinstance(folder_id, MissingType):
```

Reply:
Oh good eye, I was meaning to flag that as being probably not the cleanest way to do this, but got distracted.